.github/workflows/build_portable_linux_artifacts.yml

```diff
@@ -58,7 +58,9 @@ permissions:
 jobs:
   build_portable_linux_artifacts:
     name: Build (xfail ${{ inputs.expect_failure }})
-    runs-on: azure-linux-scale-rocm
+    # azure-linux-scale-rocm are used for regular CI builds
+    # azure-linux-scale-rocm-heavy are used for CI builds that require more resources (ex: ASAN builds)
+    runs-on: ${{ inputs.build_variant_label == 'Release' && 'azure-linux-scale-rocm' || 'azure-linux-scale-rocm-heavy' }}
```

Comment:
This condition could be flipped, so the default is `azure-linux-scale-rocm` instead of `azure-linux-scale-rocm`